runtime/coverage.go

```diff
@@ -203,31 +202,32 @@ func (r *CoverageReport) InspectProgram(location Location, program *ast.Program)
 		})
 
 	locationCoverage := NewLocationCoverage(lineHits)
-	r.Coverage[location] = locationCoverage
+	r.Coverage[location.ID()] = locationCoverage
 }
 
 // IsProgramInspected checks whether the *ast.Program on the given
 // location, has been inspected or not.
 func (r *CoverageReport) IsProgramInspected(location Location) bool {
-	_, isInspected := r.Programs[location]
+	_, isInspected := r.Programs[location.ID()]
 	return isInspected
 }
 
-// CoveredStatementsPercentage returns a string representation of
-// the covered statements percentage. It is defined as the ratio
-// of total covered lines over total statements, for all locations.
-func (r *CoverageReport) CoveredStatementsPercentage() string {
-	totalStatements := 0
-	totalCoveredLines := 0
-	for _, locationCoverage := range r.Coverage { // nolint:maprange
-		totalStatements += locationCoverage.Statements
-		totalCoveredLines += locationCoverage.CoveredLines()
-	}
-	percentage := fmt.Sprintf(
+// Percentage returns a string representation of the covered statements
+// percentage. It is defined as the ratio of total covered lines over
+// total statements, for all locations.
+func (r *CoverageReport) Percentage() string {
+	totalStatements := r.Statements()
+	totalCoveredLines := r.Hits()
+	return fmt.Sprintf(
 		"%0.1f%%",
 		100*float64(totalCoveredLines)/float64(totalStatements),
 	)
-	return fmt.Sprintf("Coverage: %v of statements", percentage)
+}
+
+// CoveredStatementsPercentage returns a human-friendly message
+// for the covered statements percentage.
+func (r *CoverageReport) CoveredStatementsPercentage() string {
```

Comment:
Maybe just implement `Stringer`, i.e. rename this function to `String()`?

Reply:
Good idea :+1: 
Updated in https://github.com/onflow/cadence/pull/2391/commits/34e424953113deeb0a5179863c9c119aefaaf1ff